packages/analytics/src/functions.ts

```diff
@@ -142,3 +148,22 @@ export async function setAnalyticsCollectionEnabled(
   const measurementId = await initializationPromise;
   window[`ga-disable-${measurementId}`] = !enabled;
 }
+
+/**
+ * Sets the variable {@link defaultEventParametersForInit} for use in the initialization of
+ * analytics.
+ *
+ * @param customParams Any custom params the user may pass to gtag.js.
+ */
+export function _setDefaultEventParametersForInit(
+  customParams: CustomParams
+): void {
+  if (defaultEventParametersForInit) {
+    defaultEventParametersForInit = {
```

Comment:
I think because this is called "set", developers will expect it to overwrite and not extend any previous object, so we don't need to extend here.